tests/integration/server/users_api_test.py

```diff
@@ -257,3 +258,72 @@ def test_user_delete_api():
         assert user == []
         assert metadata == []
         assert role_owners == []
+
+
+def test_reject_users_proposals():
+    """Test that a user's proposals are rejected when they are deleted."""
+    user_to_delete = {
+        "name": "nadia two",
+        "username": "nadia2",
+        "password": "test11",
+        "email": "nadia2@test.com",
+    }
+
+    user = {
+        "name": "nadia three",
+        "username": "nadia3",
+        "password": "test11",
+        "email": "nadia3@test.com",
+    }
+    with requests.Session() as session:
+        response1 = create_test_user(session, user_to_delete)
+        response2 = create_test_user(session, user)
+        role_payload_1 = {
+            "name": "NadiaRole1",
+            "owners": response1.json()["data"]["user"]["id"],
+            "administrators": response1.json()["data"]["user"]["id"],
+            "description": "Nadia Role 1",
+        }
+
+        role_response1 = create_test_role(session, role_payload_1)
+        proposal_2 = add_role_member(
+            session,
+            role_response1.json()["data"]["id"],
+            {"id": response2.json()["data"]["user"]["id"]},
+        )
+        next_id = response1.json()["data"]["user"]["id"]
+        conn = connect_to_db()
+        user_exists = (
+            r.db("rbac")
+            .table("users")
+            .filter({"next_id": next_id})
+            .coerce_to("array")
+            .run(conn)
+        )
+        assert user_exists
```

Comment:
Should we also also assert that the proposal currently exists here before the deletion happens?

Reply:
Disregard, as you're already checking the existence of the proposal down below.